File: week5/Hard.java
public class Hard {

    private static int collatzSteps(int n) {
        if (n == 1) return 0;

        if (n % 2 == 0) {
            return 1 + collatzSteps(n / 2);
        } else {
            return 1 + collatzSteps(3 * n + 1);
        }
    }

    public static void main(String[] args) {
        System.out.println(collatzSteps(2));
    }
}

File: week5/hard.py
def collatz_steps(n: int) -> int:
    if n == 1:
        return 0
    
    if n % 2 == 0:
        return 1 + collatz_steps(n / 2)
    else:
        return 1 + collatz_steps(3 * n + 1)

print(collatz_steps(97))

